lib/core/macros.nim

```diff
@@ -546,6 +547,9 @@ proc getAst*(macroOrTemplate: untyped): NimNode {.magic: "ExpandToAst", noSideEf
   ##     var ast = getAst(BarTemplate())
 
 proc quote*(bl: typed, op = "``"): NimNode {.magic: "QuoteAst", noSideEffect.} =
+  ## .. warning:: `quote` comes with many caveats, consider using the new
```

Comment:
If it is good enough why not just Deprecate `quote do` ?. :thinking:

Reply:
I've changed the message a bit, but deprecating `quote` can be done (or discussed) in followup PR.